py-polars/polars/functions/range/date_range.py

```diff
@@ -126,42 +122,38 @@ def date_range(
     Notes
     -----
     1) If both ``start`` and ``end`` are passed as date types (not datetime), and the
-    interval granularity is no finer than 1d, the returned range is also of
-    type date. All other permutations return a datetime Series.
+       interval granularity is no finer than 1d, the returned range is also of
+       type date. All other permutations return a datetime Series.
 
-    2) Because different months of the year have differing numbers of days, the offset
```

Comment:
the long explanation starting here is good, but to be honest I think it's way too long to be included in a docstring

I've drastically shortened it - there's already an example of how to get the last day of each month